sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -414,20 +414,12 @@ final class Decimal extends Ordered[Decimal] with Serializable {
 
   def floor: Decimal = if (scale == 0) this else {
     val newPrecision = DecimalType.bounded(precision - scale + 1, 0).precision
-    val res = toPrecision(newPrecision, 0, ROUND_FLOOR)
-    if (res == null) {
-      throw new AnalysisException(s"Overflow when setting precision to $newPrecision")
-    }
-    res
```

Comment:
So, do you want change the user facing message in the same `AnalysisException`, right?
From `Overflow when setting precision to $newPrecision` to `$toDebugString cannot be represented as Decimal($precision, $scale)`?

Reply:
Well, I did this for 2 reasons:
 1. Coherency in the exception we throw when an overflow occurs;
 2. Since we are already doing all the needed checks in `toPrecision`, checking everything again here is an overhead which can be avoided.